### PR TITLE
Upgrade operator-sdk to v1.17.0 

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

## Which problem is this PR solving?
- We would like to upgrade to operator =-sdk to v1,17 or newer
- 
## Short description of the changes
- Upgrades to operator-sdk according to https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/

### CODE REVIEW

**Comment by kevinearls** [comment]
@pavolloffay  @rubenvp8510  @esnible  Please review

**Review by pavolloffay — COMMENTED** [review]
Missing parts:
* remove support for admissionReviewVersions v1beta1
* change the golang version to 1.17 in all CI workflows and all dockerfiles 
* `--ignore-not-found` is missing in the Makefile

File: bundle/manifests/jaeger-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -23,7 +23,7 @@ metadata:
     description: Provides tracing, monitoring and troubleshooting for microservices-based
       distributed systems
     operators.openshift.io/infrastructure-features: '["disconnected"]'
-    operators.operatorframework.io/builder: operator-sdk-v1.13.0+git
+    operators.operatorframework.io/builder: operator-sdk-v1.16.0+git
```
Comment (rubenvp8510) [inline bundle/manifests/jaeger-operator.clusterserviceversion.yaml:26]: should this be 1.16 or 1.17? 🤔

File: bundle/manifests/jaeger-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -388,14 +388,20 @@ spec:
                 - --secure-listen-address=0.0.0.0:8443
                 - --upstream=http://127.0.0.1:8383/
                 - --logtostderr=true
-                - --v=10
```
Comment (rubenvp8510) [inline bundle/manifests/jaeger-operator.clusterserviceversion.yaml:391]: why this change?

File: .github/workflows/base-checks.yaml [hunk 3]
```diff
@@ -17,7 +17,7 @@ jobs:
     - name: Set up Go
       uses: actions/setup-go@v2.2.0
       with:
-        go-version: 1.16
+        go-version: 1.17
```
[the remark this answers is not in the document]
Comment (rubenvp8510) [reply .github/workflows/base-checks.yaml:20]: Is not required, but I think is a good opportunity to upgrade golang to a more recent version.

File: bundle/manifests/jaeger-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (kevinearls) [reply bundle/manifests/jaeger-operator.clusterserviceversion.yaml:26]: @rubenvp8510 I'm not sure.  This code is all generated, right?

File: bundle/manifests/jaeger-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (kevinearls) [reply bundle/manifests/jaeger-operator.clusterserviceversion.yaml:391]: Again, I don't know, this is generated code.

File: .github/workflows/base-checks.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kevinearls) [reply .github/workflows/base-checks.yaml:22]: @frzifus Actually update to 1.17 is requred: https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.17.0/

It should be in the go.mod file

**Comment by kevinearls** [comment]
@pavolloffay  yes, I had already looked for "admissionReviewVersions" and not found it, which is why I asked.

File: .github/workflows/base-checks.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (frzifus) [reply .github/workflows/base-checks.yaml:22]: Now iam confused. you are right, your go.mod file says [1.17](https://github.com/jaegertracing/jaeger-operator/pull/1813/files/17ac0d391afce6bb80f4a523a14dd228f6be5970..0fc655266338b959893753738c32916f0d35eec5#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R3). But github doesnt show a diff to the main branch. There it says [1.16](https://github.com/jaegertracing/jaeger-operator/blob/main/go.mod#L3)?

[the remark this answers is not in the document]
Comment (frzifus) [reply .github/workflows/base-checks.yaml:22]: oh damn its actually a diff between earlier commits on your branch. :facepalm: ... here is the change to [1.17](https://github.com/jaegertracing/jaeger-operator/pull/1813/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6) ...

**Comment by frzifus** [comment]
Not sure if we can merge it without fixing the `DCO Action required after 1s — DCO` thing.

I tried to realign things on https://github.com/frzifus/jaeger-operator/commits/update-operator-sdk
If you like, you can switch to that branch by:
```
git checkout main # switching to main branch
git branch -D update-operator-sdk # removing your local feature branch
git remote add frzifus git@github.com:frzifus/jaeger-operator.git # adding frzifus as source
git fetch frzifus update-operator-sdk # getting the changes
git checkout update-operator-sdk # checkout the new branch
git push origin update-operator-sdk --force # note: make sure origin points to your remote repo
```

to verify if `origin` points to your remote repo, use `git remove -v`

**Comment by kevinearls** [comment]
@pavolloffay @rubenvp8510 Can you review please and merge if it looks ok.  I think I have addressed all of your comments.  

@frzifus and I spent some time looking at the DCO failure and haven't yet figured out how to fix it but I don't think that should be a blocker.
